spec/unit/file_system_spec.rb

```diff
@@ -5,6 +5,14 @@
 describe "Puppet::FileSystem" do
   include PuppetSpec::Files
 
+  def with_file_content(content)
+    path = tmpfile('file-system')
+    file = File.new(path, 'w')
```

Comment:
I think the mode needs to be `'wb'`, otherwise ruby will strip '\r' while it's writing the test file that it tries to read back in later.